core/src/main/java/org/apache/spark/shuffle/unsafe/UnsafeShuffleWriter.java

```diff
@@ -147,8 +150,16 @@ public void write(scala.collection.Iterator<Product2<K, V>> records) throws IOEx
       closeAndWriteOutput();
       success = true;
     } finally {
-      if (!success) {
-        sorter.cleanupAfterError();
+      if (sorter != null) {
+        try {
+          sorter.cleanupAfterError();
+        } catch (Exception e) {
+          // Only throw this error if we won't be masking another
+          // error.
+          if (success) {
+            throw e;
+          }
```

Comment:
Shouldn't we log in the `else` branch to make sure that the error is never silently swallowed?